MoreLinq/Subsets.cs

```diff
@@ -108,7 +108,9 @@ public static IEnumerable<IList<T>> Subsets<T>(this IEnumerable<T> sequence, int
             // preconditions. This however, needs to be carefully considered - and perhaps
             // may change after further thought and review.
 
-            return new SubsetGenerator<T>(sequence, subsetSize);
+            return subsetSize == 0
+                ? (IEnumerable<IList<T>>) new[] {new List<T>()}
+                : new SubsetGenerator<T>(sequence, subsetSize);
```

Comment:
Since you're handling the 0 case at this level, would it make sense to restrict the usage of the `SubsetGenerator` to only allow `subsetSize > 0`?
This argument validation takes place in its constructor at line 216.

Reply:
Since the current implementation of 'SubsetGenerator' doesn't return the expected result for k = 0, yes you are totally right.
The less lazy but more risky option is to fix SubsetGenerator.

Reply:
I made SubsetGenerator handle the k=0 case